pkg/vm/vm.go

```diff
@@ -693,27 +696,48 @@ func (v *VM) execute(ctx *Context, op Instruction, parameter []byte) (err error)
 	// Numeric operations.
 	case ADD:
 		a := v.estack.Pop().BigInt()
+		v.checkBigIntSize(a)
 		b := v.estack.Pop().BigInt()
-		v.estack.PushVal(new(big.Int).Add(a, b))
+		v.checkBigIntSize(b)
+
+		c := new(big.Int).Add(a, b)
+		v.checkBigIntSize(c)
+		v.estack.PushVal(c)
 
 	case SUB:
 		b := v.estack.Pop().BigInt()
+		v.checkBigIntSize(b)
 		a := v.estack.Pop().BigInt()
-		v.estack.PushVal(new(big.Int).Sub(a, b))
+		v.checkBigIntSize(a)
+
+		c := new(big.Int).Sub(a, b)
+		v.checkBigIntSize(c)
+		v.estack.PushVal(c)
```

Comment:
We can also check value being pushed into the stack to not duplicate these checks.